packages/aws-cdk/lib/diff.ts

```diff
@@ -74,8 +76,13 @@ export enum RequireApproval {
  *
  * Returns true if the changes are prompt-worthy, false otherwise.
  */
-export function printSecurityDiff(oldTemplate: any, newTemplate: cxapi.CloudFormationStackArtifact, requireApproval: RequireApproval): boolean {
-  const diff = cfnDiff.diffTemplate(oldTemplate, newTemplate.template);
+export function printSecurityDiff(
+  oldTemplate: any,
+  newTemplate: cxapi.CloudFormationStackArtifact,
+  requireApproval: RequireApproval,
+  changeSet?: CloudFormation.DescribeChangeSetOutput,
+): boolean {
```

Comment:
This function is wrong. The input should be a `diff`, not two templates that should be diffed again. Now we're diffing them twice!

Reply:
This function is never called after `diff`, it's invoked directly by the toolkit. Could you elaborate on how the diff is happening twice?

Reply:
Hmm, I guess you're right. I must have misread.